code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/InstancePurgePostListener.java

```diff
@@ -0,0 +1,63 @@
+package io.cattle.platform.servicediscovery.process;
```

Comment:
To me, the deallocation should happen on Stop, since the InstanceStart is allocating. 
I will defer to @ibuildthecloud for why do we deallocate on Purge - but not on Instance Stop or maybe on Remove